### ISSUE TITLE
Best practices for abstraction

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/SBOL-specification#368]
We've been running into an issue around abstraction levels, with different groups representing interactions at different levels of detail. None of these is more "right" than the others, since one legitimately should be representing different details depending on the circumstance.

What is a good way, however, to handle 1) harmonization across different levels and/or 2) actually included more than one equivalent representation in a system.

Consider a simple example of this in how to represent repression of pTet by TetR:
- The TetR CDS inhibits the pTet promoter
- The TetR CDS produces TetR protein, which inhibits pTet.
- The TetR CDS produces mRNA that produces the TetR protein, which inhibits pTet.

**Comment by cjmyers** [comment]
I think these potentially can be represented as a hierarchy of abstractions, with each level down exposing more detail.

**Comment by jakebeal** [comment]
We should work out some examples and run them by people to see how they feel about them.

**Comment by jamesscottbrown** [comment]
> I think these potentially can be represented as a hierarchy of abstractions, with each level down exposing more detail.

Given a more detailed model, it should be possible to automatically do some abstraction, such as first simplifying by removing the mRNA, and then simplifying by removing the protein ([sbml-diff](https://pubs.acs.org/doi/10.1021/acssynbio.6b00273) does something like this when drawing diagrams of SBML models).

Doing the reverse is much harder: it would require TetR and pTet to be annotated with appropriate identifiers, and some external source of information that explained how they interact.

**Comment by cjmyers** [comment]
@jamesscottbrown I completely agree.  My position has always been to include as much detail as possible, but leave it to tools to do the abstraction.

**Comment by bbartley** [comment]
An interesting exercise would be to scope out whether it is possible to express in SBOL a representation that is detailed enough to support abstraction into each of the 3 levels of SBGN representation: process description, entity relationship, activity flow

**Comment by udp** [comment]
> None of these is more "right" than the others, since one legitimately should be representing different details depending on the circumstance.

Just want to +1 this while the discussion is young.  Working out how to handle different levels of abstraction is definitely something we need to do, but it is also important to remember that we cannot require any particular level of detail, even as a best practice.  The level of abstraction is part of the design and is entirely up to the designer.
